aerogear/keycloak-metrics-spi
# Pipeline for releasing Docker init container

Hi, I've created Github action based pipeline that allows to promote release by Github release [form](https://github.com/aerogear/keycloak-metrics-spi/releases/new). It adds built jar file into Github release and also creates and publishes init container into Docker hub. To make this 100% working you'd need to have account on Docker hub and create repo for this project, then create token that will allow the pipeline to publish image into Docker hub. Token from Docker hub should be added to this Github repo as repository secret called `DOCKER_HUB_TOKEN`. Working test release can be found in my fork [here](https://github.com/kilmajster/keycloak-metrics-spi/runs/4000264666?check_suite_focus=true)
If you need more details about configuration, just let me know :)

Regards, Łukasz Włódarczyk

## Motivation
#34

## What
Pipeline that publishes jar into Github releases and init container into Docker hub

## Why
To allow using this provider with Helm charts

## How
Release process with this pipeline looks like following:
1. You create new tag using standard Github release mechanism, e.g. tag = `2.5.3`
2. Pipeline is executing tests
3. Pipeline sets project version based on git tag and build jar
5. Jar file is added into Github release
6. Jar file is added into Docker image
7. Docker image is tagged same as Github tag and also as `latest`
8. Docker images are pushed into Docker hub

## Verification Steps
You can verify that jar is added into Github release in my fork [here](https://github.com/kilmajster/keycloak-metrics-spi/releases/tag/2.5.3)  
To verify if docker container contains jar file you can do following:
```bash
docker run --rm -it --entrypoint sh kilmajster/keycloak-metrics-spi

# then into docker container
 cd keycloak-metrics-spi
 ls -al
 
 # and also for eg. to verify if version is set correctly
 
 unzip keycloak-metrics-spi-2.5.3.jar
 
 ...
 
 cat META-INF/MANIFEST.MF
```

## Checklist:


## Progress

## Additional Notes
